### PR TITLE
Add credential field to discoveryconfig

### PR DESCRIPTION
**Pull request by JakobGray**
User story: https://github.com/open-cluster-management/backlog/issues/11473

Replace ProviderConnections field with Credential field, accepting a singular secret name. Remove support for multiple connections. This is to work towards enabling better RBAC control for DiscoveredCluster visibility.

### CODE REVIEW

File: api/v1/discoveryconfig_types.go [hunk 1]
```diff
@@ -38,6 +38,10 @@ type DiscoveryConfigSpec struct {
 	// +optional
 	ProviderConnections []string `json:"providerConnections,omitempty"`
 
+	// Credential is the secret containing credentials to connect to the OCM api on behalf of a user
+	// +optional
```
Comment (zkayyali812) [inline api/v1/discoveryconfig_types.go:42]: Perhaps this should be mandatory? 
Is there a point to a discoveryconfig w/o a credential?

Comment (JakobGray) [reply api/v1/discoveryconfig_types.go]: Removed optional flag

**Review by zkayyali812 — APPROVED** [review]

**Pull request merged**